crates/binstalk/src/helpers/git.rs

```diff
@@ -53,7 +74,7 @@ impl FromStr for GitUrl {
 }
 
 #[derive(Debug)]
-pub struct Repository(gix::Repository);
+pub struct Repository(pub(crate) gix::Repository);
 
 impl Repository {
     /// WARNING: This is a blocking operation, if you want to use it in
```

Comment:
NOTE that you would also need to modify use of [`load_manifest_from_workspace`](https://github.com/cargo-bins/cargo-binstall/blob/963e9e97add0a681f117fe21c6b3fde074706ac7/crates/binstalk/src/helpers/cargo_toml_workspace.rs#L20) in [`resolve.rs`](https://github.com/cargo-bins/cargo-binstall/blob/963e9e97add0a681f117fe21c6b3fde074706ac7/crates/binstalk/src/ops/resolve.rs#L374) since it also uses `Repository::shallow_clone` here.